core/libp2p/multi/c-utils/protoutils.c

```diff
@@ -354,26 +367,26 @@ char *bytes_to_string(char **buffer,
       unsigned char b58[b58_size];
       memset(b58, 0, b58_size);
       unsigned char *ptr_b58 = b58;
-      int returnstatus = multiaddr_encoding_base58_encode(
-          addrbuf, num_bytes, &ptr_b58, &b58_size);
+      int returnstatus = multiaddr_encoding_base58_encode(addrbuf, num_bytes,
+                                                          &ptr_b58, &b58_size);
       free(addrbuf);
       if (returnstatus == 0) {
         char *error = (char *)malloc(sizeof(char) * (38 + addrsize + 1));
-        strcpy(error, "unable to base58 encode MultiAddress: ");
-        strcat(error, IPFS_ADDR);
+        strcpy(error, "unable to base58 encode MultiAddress: ");  // NOLINT
+        strcat(error, IPFS_ADDR);                                 // NOLINT
         return error;
       }
-      strcat(results, "/");
-      strcat(results, protocol->name);
-      strcat(results, "/");
-      strcat(results, (char *)b58);
+      strcat(results, "/");             // NOLINT
+      strcat(results, protocol->name);  // NOLINT
+      strcat(results, "/");             // NOLINT
+      strcat(results, (char *)b58);     // NOLINT
     }
   } else {
-    char *error = (char *)malloc(sizeof(char) * (27));
-    strcpy(error, "unable to decode a protocol");
+    char *error = (char *)malloc(sizeof(char) * (27));  // NOLINT
+    strcpy(error, "unable to decode a protocol");       // NOLINT
     return error;
```

Comment:
Aren't memory leaks in this function as well?

Reply:
yes, c-utils is a complete mess, it is better to get rid of it